serialization.rst

```diff
@@ -222,6 +225,67 @@ possible mapping values are:
   if it exists, will be sent as an associative array to the ``bsonUnserialize``
   function to initialise the object's properties.
 
+Scalar Types
+------------
+
+For types that the driver has a wrapper class for, it is possible to define a
+map from each supported data type to a class of your own, providing it
+implements the ``MongoDB\BSON\TypeWrapper`` interface.
+
+The supported data types are Binary, Decimal128, Javascript, MaxKey, MinKey,
+ObjectID, Regex, Timestamp, and UTCDateTime.
+
+The ``MongoDB\BSON\TypeWrapper`` interface defines two functions:
+``createFromBSONType()``, a factory method which takes a ``MongoDB\BSON\Type``
+argument, and ``toBSONType()``.
+
+As an example, a wrapped UTCDateTime class, could look like::
+
+    class UTCDateTimeWrapper implements \MongoDB\BSON\TypeWrapper
+    {
+        private $intern;
+
+        public function __construct( \MongoDB\BSON\UTCDateTime $type )
+        {
+            $this->intern = $type->toDateTime();
+        }
+
+        static function createFromBSONType(\MongoDB\BSON\Type $type)
+        {
+            if (! $type instanceof \MongoDB\BSON\UTCDateTime) {
+                throw UnexpectedValueException;
+            }
+
+            return new UTCDateTimeWrapper( $type );
+        }
+
+        function toBSONType()
+        {
+            return new \MongoDB\BSON\UTCDateTime( $this->intern );
+        }
+    }
+
+
+If the defined class wraps (composes) an original ``MongoDB\BSON\*`` class,
+then it SHOULD also implement the accompanying
+``\MongoDB\BSON\<classname>Interface`` interface.
+
+The type specific interfaces include all the type-specific methods from the
+original class, with the exact same arguments and return types. For example,
+they will not include the ``__construct()`` and ``__debugInfo()`` methods.
+
+For example, a user-defined ``UTCDateTimeWrapper`` class needs to implement the
+``MongoDB\BSON\UTCDateTimeInterface`` and ``MongoDB\BSON\TypeWrapper``
+interfaces.
+
+It is not necessary to return an object from the ``createFromBSONType``
+"factory" method. You may also return a simple scalar value. You could use
+this as a way to migrate Decimal128 values to plain strings, or downgrade them
+to floating points or integers. Likewise, a UTCDateTime wrapper that converts
```

Comment:
This sentence is missing a subject and predicate. How about: "you could implement a UTCDateTime wrapper that converts dates to a formatted string".